### DISCUSSION TITLE
When I try to authenticate using Github, I get a "User not found" message.

### DISCUSSION THREAD

**Discussion by motsuo373** [liveblocks/liveblocks#908] [Help] [answered]
Although I set it up by looking at the GitHub of the guide at the following URL,
I get an error message "User not found".
https://liveblocks.io/docs/guides/nextjs-starter-kit#github-authentication

<img width="583" alt="image" src="https://github.com/liveblocks/liveblocks/assets/62975702/6ed41400-b4a2-4437-978b-bc05299952c9">

When I press the GitHub authentication button, I am returned to the authentication page.
(The google button is something I added. What I want to check is GitHub.)
![image](https://github.com/liveblocks/liveblocks/assets/62975702/36e0d96f-4093-45e8-94b7-8e0301791642)

At first I thought it was due to missing data in `data/users.ts`.
I put in my own data, but it was undefined.

<img width="440" alt="貼り付けた画像_2023_05_28_2_04" src="https://github.com/liveblocks/liveblocks/assets/62975702/730ac516-9cc0-4929-9ab3-07eec3c41af7">

Upon checking, I noticed that the id was undefined.

Am I doing something wrong around authentication at this time?
I checked the Client secrets on github and it says it is being used, so I am assuming the authentication is correct as well.

Or is it possible that you made some mistake in your local environment?
I would like to solve this problem and use it for my own product.

Sorry for my poor English and basic questions.

**Comment by CTNicholas** [comment, marked as answer]
Hey @motsuo373! Sorry for the slow response—this slipped through the cracks. I've just tried setting up a new starter kit app following the GitHub auth guide, and it's working okay for me. 

Have you modified any other code in the starter kit? If so, feel free to share any code so that we can help you fix it. The error you're getting suggests that `session.user.email` is `undefined` on [this line](https://github.com/liveblocks/liveblocks/blob/6fa3640952a2a38805359aae64f82744ce3f568e/starter-kits/nextjs-starter-kit/pages/api/auth/%5B...nextauth%5D.ts#L16). This is a [NextAuth callback](https://next-auth.js.org/configuration/callbacks#session-callback), and `user.email` [should not be undefined](https://next-auth.js.org/getting-started/client).

Have you tried using a different browser or a different email? Bear in mind that the email must have a GitHub account linked to it. Feel free to share more info, and I'll try to help 🙂.

**Reply by motsuo373** [reply]
@CTNicholas 
Thenk you for reply! 
I was checking and realized that the problem is that the github email address is not done correctly.


```
export const authOptions = {
  secret: NEXTAUTH_SECRET,
  callbacks: {
    // Get extra user info from your database to pass to front-end
    // For front end, update next-auth.d.ts with session type
    async session({ session }: { session: any }) {
      console.log(session); // check here
      const userInfo: User | null = await getUser(session.user.email);

~~~~

  // Configure one or more authentication providers
  // More info: https://next-auth.js.org/providers/
  providers: [
    // Use GitHub authentication
    GithubProvider({
      clientId: process.env.GITHUB_CLIENT_ID as string,
      clientSecret: process.env.GITHUB_CLIENT_SECRET as string,
    }),

    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID as string,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
    }),
  ],
};
```

I looked at the contents of the session and found that the email was undefined.
That is why I noticed this problem.

```
  {
    user: { name: 'my-name', email: undefined, image: 'https://avatars.githubusercontent.com/u/62975702?v=4' },
    expires: '2023-07-15T11:44:49.629Z'
  }
```

I would like to review my email verification settings.
thanks.
